research/m-out-of-n.md

```diff
@@ -0,0 +1,44 @@
+Let say we have input data. And we need to create `n` data sets that:
+- having any `m` of data sets, input data can be recovered,
+- having less than `m` of data sets, input data cannot be recovered.
+
+Resulting data sets are understood as user secrets and input data is used to compute a private key:
+```script
+pk = keccak256(input_data)
```

Comment:
I don't think it is always true, because we have:
- `pk = keccak256(input_data)` for system generated pk
- `pk = reverse_ecdsa(input_data)` for user-provided pk

I would skip any details of computation and simply say that "input data is used to compute a private key"